node-js-basics-101/node_modules/asynquence-contrib/plugin.pThenCatch.js

```diff
@@ -0,0 +1,123 @@
+// "pThen"
+ASQ.extend("pThen",function $$extend(api,internals){
+	return function $$pthen(success,failure) {
+		if (internals("seq_aborted")) {
+			return api;
+		}
+
+		var ignore_success_handler = false, ignore_failure_handler = false;
+
+		if (typeof success === "function") {
+			api.then(function $$then(done){
+				if (!ignore_success_handler) {
+					var ret, msgs = ASQ.messages.apply(ø,arguments);
+					msgs.shift();
+
+					if (msgs.length === 1) {
+						msgs = msgs[0];
+					}
+
+					ignore_failure_handler = true;
+
+					try {
+						ret = success(msgs);
+					}
+					catch (err) {
+						if (!ASQ.isMessageWrapper(err)) {
+							err = [err];
```

Comment:
__[jshint]__
* Do not assign to the exception parameter.

<!-- Comment by [SideCI](https://sideci.com) -->